lms/djangoapps/course_api/permissions.py

```diff
@@ -2,9 +2,65 @@
 Course API Authorization functions
 """
 
+from django.contrib.auth.models import User
+from rest_framework import permissions
+
 from student.roles import GlobalStaff
 
 
+class MasqueradingPermission(permissions.BasePermission):
+    """
+    Allow authorized users to masquerade as other users, taking on that
+    user's permissions.
+
+    This class must be listed before any other Permission classes
+    to properly enable masquerading
+
+    To masquerade as a user, add a query parameter to the URL specifying the
+    masqueraded user's username:
+
+        GET https://courses.edx.org/api/courses/v1/courses?masquerade=henry
+
+    To use a parameter other than `masquerade`, set `masquerading_param` as a
+    class attribute on any view that uses `MasqueradingPermission`.
+    """
+
+    default_masquerading_param = 'masquerade'
```

Comment:
It seems that our standard value for this is "username".  Probably worth updating this to "username", but I have some concern that "username" is a pretty generic and non-descriptive term for this, that might end up getting overloaded with other meanings.